src/crl.c

```diff
@@ -99,8 +99,25 @@ static int InitCRL_Entry(CRL_Entry* crle, DecodedCRL* dcrl, const byte* buff,
         return WOLFSSL_FAILURE;
     }
 #endif
-    crle->certs = dcrl->certs;   /* take ownsership */
+#ifdef CRL_USE_CERT_BUFFERS
+    int totalCerts = dcrl->totalCerts;
+    RevokedCert* rc = dcrl->certs;
+
+    if (totalCerts >= CRL_MAX_REVOKED_CERTS)
+        return WOLFSSL_FAILURE;
+
+    while (totalCerts > 0) {
+        /* revoked cert chains were placed in reverse order in GetRevoked()
+         * copy the first cert in chain into the last array position */
+        if (rc != NULL) {
```

Comment:
Maybe put the `rc != NULL` in the while expression? Otherwise you could enter an endless loop here if rc == NULL at start.